src/ansys/fluent/core/streaming_services/events_streaming.py

```diff
@@ -277,14 +552,16 @@ def _register_solution_event_sync_callback(
         """
         )
 
-        def on_pause(session, event_info: EventsProtoModule.AutoPauseEvent):
-            if unique_id == event_info.level:
-                event_info_cls = (
-                    EventsProtoModule.TimestepEndedEvent
-                    if event_type == SolverEvent.TIMESTEP_ENDED
-                    else EventsProtoModule.IterationEndedEvent
-                )
-                event_info = event_info_cls(index=event_info.index)
+        def on_pause(session, event_info: SolutionPausedEventInfo):
+            if unique_id == int(event_info.level):
+                if event_type == SolverEvent.ITERATION_ENDED:
+                    event_info = IterationEndedEventInfo(index=event_info.index)
+                else:
+                    event_info = TimestepEndedEventInfo(
+                        # TODO: Timestep size is currently not available
```

Comment:
We are casting a `SolutionPausedEventInfo` into `TimestepEndedEventInfo`. Currenlty there is no way to serialize time-step size to `EventsProtoModule.AutoPauseEvent`. We need to change the proto - will look into separately.